File: src/core/framework/IService.ts
export enum ServiceState {
  UNINITIALIZED = "UNINITIALIZED",
  INITIALIZING = "INITIALIZING",
  RUNNING = "RUNNING",
  PAUSED = "PAUSED",
  STOPPED = "STOPPED",
  FAILED = "FAILED"
}

export interface ServiceHealth {
  status: "healthy" | "degraded" | "unhealthy";
  message: string;
  timestamp: Date;
  details?: Record<string, any>;
}

export interface IService {
  readonly id: string;
  readonly name: string;
  readonly dependencies: string[];
  readonly state: ServiceState;

  init(): Promise<void>;
  start(): Promise<void>;
  stop(): Promise<void>;
  healthCheck(): Promise<ServiceHealth>;
}

File: NovaSim/NovaSim.Core/Interfaces/IService.cs
using System;
using System.Threading.Tasks;
using NovaSim.Core.Enums;

namespace NovaSim.Core.Interfaces
{
    public record ServiceHealth(string Status, string Message, DateTime Timestamp, Dictionary<string, object>? Details = null);

    public interface IService
    {
        string Id { get; }
        string Name { get; }
        string[] Dependencies { get; }
        ServiceState State { get; }

        Task InitAsync();
        Task StartAsync();
        Task StopAsync();
        Task<ServiceHealth> HealthCheckAsync();
    }
}
